posts/multiprocessing-vs-multithreading-python.md

```diff
@@ -0,0 +1,137 @@
+
+# Multiprocessing vs. Multithreading in Python: What you need to know.
+
+***TLDR:*** If you don't want to understand the under-the-hood explanation, here's what you've been waiting for: you can use `threading` if your program is network bound or you are running your UI in Python using something like TkInter and `multiprocessing` if it's CPU bound.
+
+_Just a disclaimer: we're a logging company here @ Timber. We'd love it if you tried out [our product](timber.io) (it's seriously great!), but that's all we're going to advertise it ... you guys came here to learn about the difference between threading and multiprocessing in python, and this guide won't disappoint._
+
+We're creating this guide because when we went looking for the difference between threading and multiprocessing, we found the information out there unnecessarily difficult to understand. They went far too in-depth, without really touching the meat of the information that would help us decide what to use and how to implement it.
+
+## What is Threading and why do you want it
+
+By nature, Python is a linear language, but the threading module comes in handy when you want a little more processing power. While threading in Python cannot be used for parallel CPU computation, it's perfect for I/O operations such as web scraping because the processor is sitting idle waiting for data.
+
+Threading is game-changing because many scripts related to network/data I/O spend the majority of their time waiting for data from a remote source. Because downloads might not be linked (i.e., scraping separate websites), the processor can download from different data sources in parallel and combine the result at the end. For CPU intensive processes, there is little benefit to using the threading module.
+
+![Threading Architecture](/images/multiprocessing-vs-multithreading-python/threadingSameDataspace.jpeg)
```

Comment:
Should this be a contentful image?

Reply:
Yup, hadn't uploaded to Contentful yet & wanted to create a system for freelancers to add photos to the Github repo.